airflow/ui/src/components/ClearRun/ClearRunTaskAccordion.tsx

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Text } from "@chakra-ui/react";
+import { Link } from "@chakra-ui/react";
+import type { ColumnDef } from "@tanstack/react-table";
+import { Link as RouterLink } from "react-router-dom";
+
+import type {
+  TaskInstanceCollectionResponse,
+  TaskInstanceResponse,
+} from "openapi/requests/types.gen";
+import { DataTable } from "src/components/DataTable";
+import { Status } from "src/components/ui";
+
+import { Accordion } from "../ui";
+
+const columns: Array<ColumnDef<TaskInstanceResponse>> = [
+  {
+    accessorKey: "task_display_name",
+    cell: ({ row: { original } }) => (
+      <Link asChild color="fg.info" fontWeight="bold">
+        <RouterLink
+          to={`/dags/${original.dag_id}/runs/${original.dag_run_id}/tasks/${original.task_id}${original.map_index > -1 ? `?map_index=${original.map_index}` : ""}`}
+        >
+          {original.task_display_name}
+        </RouterLink>
+      </Link>
+    ),
+    enableSorting: false,
+    header: "Task ID",
+  },
+  {
+    accessorKey: "state",
+    cell: ({
+      row: {
+        original: { state },
+      },
+    }) => <Status state={state}>{state}</Status>,
+    enableSorting: false,
+    header: () => "State",
+  },
+  {
+    accessorKey: "map_index",
+    enableSorting: false,
+    header: "Map Index",
+  },
+
+  {
+    accessorKey: "dag_run_id",
+    enableSorting: false,
+    header: "Run Id",
+  },
+];
+
+type Props = {
+  readonly affectedTasks?: TaskInstanceCollectionResponse;
+};
+
+// Table is in memory, pagination and sorting are disabled.
+// TODO: Make a front-end only unconnected table component with client side ordering and pagination
+const ClearRunTasksAccordion = ({ affectedTasks }: Props) => (
+  <Accordion.Root collapsible>
+    <Accordion.Item key="tasks" value="tasks">
+      <Accordion.ItemTrigger>
+        <Text fontWeight="bold">
+          Affected Tasks: {affectedTasks?.total_entries ?? 0}
+        </Text>
+      </Accordion.ItemTrigger>
+      <Accordion.ItemContent>
+        <Box maxH="400px" overflowY="scroll">
+          <DataTable
```

Comment:
Interesting - if no items are found (DAG run was success and you select failed tasks only) then it is displaying "no task instances found" but below that text still the table headers are shown.

Reply:
I agree that in this particular case maybe hiding the table all together would be good. But for other places (like tables for browsing, I think keeping it might be clearer for the user).

I'm not taking any decision here. Keeping it like that for now but If we guys agree that this should be gone, it's a one liner change.